core/auth/domain/auth.go

```diff
@@ -0,0 +1,78 @@
+package domain
+
+import (
+	"encoding/gob"
+	securityDomain "flamingo.me/flamingo/v3/core/security/domain"
+)
+
+type (
+	// Idendity information
+	Idendity interface {
+		User() User
+	}
+
+	// User is a basic authenticated user
+	User interface {
+		Subject() string
+		Email() (string,bool)
+		Name() (string,bool)
+		CustomField(string) (string,bool)
+		Roles() []securityDomain.Role
+	}
+
+	//SimpleUser - implements User and can be used
```

Comment:
As we will use godoc please format the comments like
```
// SimpleUser is a default implementation of the User interface...
```
and not
```
//SimpleUser - implements User ...
```
https://golang.org/doc/effective_go.html#commentary